add other queries and fill analytics section (#193)
timescale/docs: timescaledb/tutorials/nfl-fantasy-league.md

## Analyze NFL data

Now that you have all the data ingested, let's go over some ideas on how you can analyze the data using PostgreSQL and TimescaleDB to help you perfect
your fantasy drafting strategy and win your fantasy season.

> Some of the analysis below has some visualizations included to help you see your analysis in action. These visualizations wer created using the Matplotlib Python module, this being one of many great visualization tools. 

Reviewer: ```suggestion
Some of this analysis includes visualizations to help you see your analysis in action. These are created using the Matplotlib Python module, which is one of many great visualization tools. 
```

Also reflow.

> For optimized analysis, you will need to create the following continuous aggregate view. By using continuous aggregates, we significantly cut down on query run time (up to 30 times faster), thus making this analysis and other possible queries more easily accessible. This coninuous aggregate sums all the players movement in yards over one day and groups them by the players ID and game ID. 

Reviewer: ```suggestion
To optimize the analysis, you need to create a continuous aggregate view. Continuous aggregates significantly cut down on query run time, running up to thirty times faster, which makes this analysis and other possible queries more easily accessible. This continuous aggregate sums all the players movement in yards over one day and groups them by the players ID and game ID. 
```

And reflow.

```sql
CREATE MATERIALIZED VIEW player_yards_by_game
WITH (timescaledb.continuous) AS
SELECT t.player_id, t.gameid, 
	time_bucket(INTERVAL '1 day', t."time") AS bucket,
	SUM(t.dis) AS yards
FROM tracking t 
GROUP BY t.player_id, t.gameid, bucket;
```

1. [Number of yards run in game for passing plays, by player and game](#number-of-yards-run-in-game-for-passing-plays-by-player-and-game)
1. [Average yards run for a player over a game](#average-yards-run-for-a-player-over-a-game)
1. [Average and median yards run per game by type of player (not taking avg of individual)](#average-and-median-yards-run-per-game-by-type-of-player-not-taking-avg-of-individual)
1. [Number of snap plays by player where they were on the offense](#number-of-snap-plays-by-player-where-they-were-on-the-offense)
1. [Number of plays vs points scored](#number-of-plays-vs-points-scored)
1. [Average yards per game for top three players of each position](#average-yards-per-game-for-top-three-players-of-each-position)

### **Number of yards run in game for passing plays, by player and game**

> To get this data, use the following query which pulls yard data from the continuous aggregate and then joins that on the player table to get player details. 

Reviewer: ```suggestion
Use this query to get the yard data from the continuous aggregate. You can then join that on the player table to get player details. 
```

```sql
SELECT a.player_id, display_name, SUM(yards) AS yards, gameid 
FROM player_yards_by_game a
LEFT JOIN player p ON a.player_id = p.player_id 
GROUP BY a.player_id, display_name, gameid 
ORDER BY gameid ASC, display_name
```

> Your data will look like this:

Reviewer: ```suggestion
Your data should look like this:
```

|player_id| display_name        | yards           | gameid  |
|-----| ------------- |:-------------:| -----:|
|2555415| Austin Hooper     | 765.52 | 2018090600 |
|2556445| Brian Poole    | 661.74     |   2018090600 |

> |2560854| Calvin Ridley | 822.3     |    2018090600 |

Reviewer: Remove whitespace.

> This query can be the foundation of many other analysis questions. In the following queries, we will be returning to the query for further analysis. 

Reviewer: ```suggestion
This query can be the foundation of many other analysis questions. This section returns to the query for further analysis. 
```

### **Average yards run for a player over a game**

> This query uses one of timescale's percentile functions to find the mean yards run per game by a single player. 

Reviewer: ```suggestion
This query uses one of the TimescaleDB percentile functions to find the mean yards run per game by a single player. 
```

```sql
WITH sum_yards AS (
  SELECT a.player_id, display_name, SUM(yards) AS yards, gameid 
  FROM player_yards_by_game a
  LEFT JOIN player p ON a.player_id = p.player_id 
  GROUP BY a.player_id, display_name, gameid 
)
SELECT player_id, display_name, mean(percentile_agg(yards)) as yards
FROM sum_yards
GROUP BY player_id, display_name
ORDER BY yards DESC
```

> After running this query you will notice the player_id and display_name are null for the first row. Can you guess what this row of data means? What could the average number of yards run mean for the top players listed?

Reviewer: ```suggestion
When you run this query you might notice the `player_id` and `display_name` are null for the first row. Can you guess what this row of data means? What could the average number of yards run mean for the top players listed?
```
And reflow

I'm not sure about the hypothetical questions here. Can we just declare why it's so instead?

### **Average and median yards run per game by type of player (not taking avg of individual)**

> In this query you need to add poistion information to the sum_yards table. Once you add this column, you then can group the mean yards and median by position. For this query you will also utilize another one of timescale's ppercentile functions percentile_agg. For this query we are using this function with the 50th percentile in order to get the data's approximate median values. 

Reviewer: ```suggestion
In this query you need to add position information to the `sum_yards` table. When you add this column, you can group the mean yards and median by position. For this query you also use another one of the TimescaleDB percentile functions, called `percentile_agg`. For this query we are using `percentile_agg` to find the 50th percentile, to get the approximate median value. 
```
And reflow

```sql
WITH sum_yards AS (
  SELECT a.player_id, display_name, SUM(yards) AS yards, p.position, gameid 
  FROM player_yards_by_game a
  LEFT JOIN player p ON a.player_id = p.player_id 
  GROUP BY a.player_id, display_name, p.position, gameid 
)
SELECT position, mean(percentile_agg(yards)) AS mean_yards, approx_percentile(0.5, percentile_agg(yards)) AS median_yards
FROM sum_yards
GROUP BY position
ORDER BY mean_yards DESC
```
If you scroll to the bottom of your results you should see this:

|position| mean_yards        | median_yards  |
|-----| ------------- |:----------------:|
|HB| 275.04279069767404    | 250.88667462709043 |
|DE| 185.76162011173133   | 33.750683636185684 |
|FB| 100.37912844036691 | 67.0876116670915 |
|DT| 19.692499999999992  | 17.796475991050432 |

> Notice how the Defensive End (DE) position has such a large discrepency between the mean and median values. The median data implies that generally most DE players do not run very much during passing plays. However, the mean data shows that some of the DE players must be running a lot. Who could those players be?

Reviewer: Again on the hypothetical, and it needs to be reflowed.

### **Number of snap plays by player where they were on the offense**

> For this next query, you will be counting the number of passing events a player was involved in while playing the offensive. You will notice how much slower this query runs than the ones above which use continous aggregates. The speed you see here is comprable to what you would get in the other queries without using continous aggregates. 

Reviewer: ```suggestion
In this query, you are counting the number of passing events a player was involved in while playing the offensive. You will notice how much slower this query runs than the ones above which use continuous aggregates. The speed you see here is comparable to what you would get in the other queries without using continuous aggregates. 
```

```sql
WITH snap_events AS (
-- Create a table that filters the play events to show only snap plays
-- and display the players team information
	SELECT DISTINCT player_id, t.event, t.gameid, t.playid,
		CASE 
			WHEN t.team = 'away' THEN g.visitor_team 
			WHEN t.team = 'home' THEN g.home_team 
			ELSE NULL 
			END AS team_name
	FROM tracking t 
	LEFT JOIN game g ON t.gameid = g.game_id 
	WHERE t.event LIKE '%snap%'
)
-- Count these events and filter results to only display data when the player was
-- on the offensive
SELECT a.player_id, pl.display_name, COUNT(a.event) AS play_count, a.team_name
FROM snap_events a
LEFT JOIN play p ON a.gameid = p.gameid AND a.playid = p.playid 
LEFT JOIN player pl ON a.player_id = pl.player_id 
WHERE a.team_name = p.possessionteam 
GROUP BY a.player_id, pl.display_name, a.team_name
ORDER BY play_count DESC
```

> Notice that the two highest passing play counts are for Ben Roethlisberger and JuJu Smith-Schuster, a Quarterback and Wide Reciever respectively for the Pittburgh Steelers. Could they be some of the top offensive choices for your league?

Reviewer: ```suggestion
Notice that the two highest passing play counts are for Ben Roethlisberger and JuJu Smith-Schuster, a Quarterback and Wide Receiver respectively for the Pittsburgh Steelers. Could they be some of the top offensive choices for your league?
```

Hypothetical, plus reflow.
